core/git_command.py

```diff
@@ -245,38 +254,40 @@ def _repo_path(self, throw_on_stderr=True):
         Return the absolute path to the git repo that contains the file that this
         view interacts with.  Like `file_path`, this can be overridden by setting
         the view's `git_savvy.repo_path` setting.
-        """
-        def invalid_repo():
-            if throw_on_stderr:
-                raise ValueError("Unable to determine Git repo path.")
-            return None
 
+        Do not raise error when throw_on_stderr is `False`. It is needed
+        in GsUpdateStatusBarCommand, otherwise, spurious popup will be shown.
+        """
         # The below condition will be true if run from a WindowCommand and false
         # from a TextCommand.
         view = self.window.active_view() if hasattr(self, "window") else self.view
+        window = view.window()
         repo_path = view.settings().get("git_savvy.repo_path")
 
         if not repo_path or not os.path.exists(repo_path):
-            working_dir = self.find_working_dir()
-            if not working_dir or not os.path.isdir(working_dir):
-                return invalid_repo()
-
-            stdout = self.git(
-                "rev-parse",
-                "--show-toplevel",
-                working_dir=working_dir,
-                throw_on_stderr=throw_on_stderr
-                )
+            file_name = view.file_name()
 
-            repo_path = stdout.strip()
+            # try the current file first
+            if file_name:
+                repo_path = self.find_git_toplevel(
+                    os.path.dirname(file_name), throw_on_stderr=False)
+                # only set `git_savvy.repo_path` for current file
+                if repo_path:
+                    view.settings().set("git_savvy.repo_path", os.path.realpath(repo_path))
 
+            # fallback: use the first folder if the current file is not inside a git repo
             if not repo_path:
-                return invalid_repo()
+                if window:
+                    folders = window.folders()
+                    if folders:
+                        repo_path = self.find_git_toplevel(
+                            folders[0], throw_on_stderr=throw_on_stderr)
```

Comment:
Image there are two windows, W1 and W2. Each of them has a folder of git repo.
Now you run `subl .bashrc` and it opens at W1

While editing the `.bashrc` file on W1, you may still use the repo of W1. But we don't want to set the `git_savvy.repo_path` to the first folder of W1 incase you want to move `.bashrc` to W2 and continue editing.